src/Router/buildClientApp.tsx

```diff
@@ -61,25 +66,38 @@ export function buildClientApp(config: AppConfig): Promise<ClientResolveProps> {
         render,
       })
 
-      const bootProps = {
-        system: {
-          ...config,
-          relayEnvironment,
-          resolver,
-          routes,
-          currentUser,
-        },
-      }
-
       try {
         await loadComponents()
       } catch (error) {
         // FIXME: https://github.com/smooth-code/loadable-components/pull/93
       }
 
-      const ClientApp = props => {
+      const system: Router = {
+        relayEnvironment,
+        resolver,
+        routes,
+        currentUser,
+      }
+
+      const ClientApp: App = props => {
         return (
-          <Boot {...bootProps} {...props}>
+          <Boot
+            system={system}
+            initialBreakpoint={initialBreakpoint}
+            initialState={[
+              new AppState({ ...initialAppState, system }),
+              ...initialState,
```

Comment:
We need to move `system` and `mediator` out of the state and into the `Artsy` context, which is static. Once that’s done the `initialAppState` prop and `AppState` can be removed.